cli/collection_create.go

```diff
@@ -16,34 +16,41 @@ import (
 
 	"github.com/spf13/cobra"
 
+	acpIdentity "github.com/sourcenetwork/defradb/acp/identity"
 	"github.com/sourcenetwork/defradb/client"
 )
 
 func MakeCollectionCreateCommand() *cobra.Command {
+	const identityFlagLongRequired string = "identity"
+	const identityFlagShortRequired string = "i"
+
+	var identityValue string
 	var file string
+
 	var cmd = &cobra.Command{
-		Use:   "create <document>",
+		Use:   "create [-i --identity] <document>",
```

Comment:
todo: I don't think we should be duplicating this property manually for every command, it should be inherited by all commands like collection is/was(?) (for collection commands).  If we keep this param.

Reply:
I am happy to inherit it through the collection if it's possible to do so in a simple manner. Do you want it to be added to the `client.CollectionFetchOptions` or be a separate thing?

This param should def stay even when we have the capability to set it in the config file.

Reply:
It should be a separate thing, an may need to be done/present when collections are not involved.

Reply:
For now it is only allowed as an optional arg on (4/6) collection options, and it is specified there, in all other places it is invalid (with the exception of `request`).

For now, I am not in favour of allowing the `--identity` flag on places where it makes no sense or isn't used at all. Can look into making it global, when the config setting functionality is added.  

It's not used / useless on the last two collection commands (`docIDs` and `describe`):
```
  create      Create a new document.
  delete      Delete documents by docID or filter.
  get         View document fields.
  update      Update documents by docID or filter.
  
  describe    View collection description.
  docIDs      List all document IDs (docIDs).
```

Reply:
How is it not to be used for `docIDs`?  If a user has no access to a collection I don't think we should be returning doc ids from it.

Long-term tracking which actors call `describe` (or any other commands), plus some ACP stuff is likely to be desired, and I would rather have the param's consistency enforced by architecture than exclude it from those two commands (if you cant find another way to enforce this).

Reply:
1) All docIDs are public (so are foreignIDs, even if the document they point to is private).
2) Access is **not** on collection but the document. So there is no such this as "no access to collection" for now. 

> consistency enforced by architecture

Even if we want it globally, that is a separate feature (likely to be done after/with the config file stuff).

Reply:
> Even if we want it globally

It doesn't have to be global - add it to the collection command

Reply:
> All docIDs are public (so are foreignIDs, even if the document they point to is private).

Their membership to a given collection is private.

Reply:
> Access is not on collection but the document. So there is no such this as "no access to collection" for now.

The below quote in the PR description makes it sound like there is collection level ACP - can you please clarify?

> The collection has a permission on it (i.e. collection is permissioned).

Reply:
> It doesn't have to be global - add it to the collection command

Is there a way to limit inheritance to only the commands that need the `identity` flag? Would like to limit users from specifying it in places where it is a `dummy` in.

> Their membership to a given collection is private. So is their membership to a node, unless ACP inherently has to leak that due to some design decision that I do not know of/have forgotten.
>
> This feels very wrong. If a document is owned and covered by a policy, other actors without permission to read it should not be able to read any of it, including its ID and which collection/nodes this belongs to.

I mostly agree with you, and happy to follow what the consensus is. Me and John did decide to think of docIDs as public because any document can technically link to another document that might be private (as all fields of the document are owned by the owner, including the foreign key). Consider a document owned by `actor1` but it specifies a foreignID linking to a document that is not permissible for `actor1` to access. In that case, even though the document that is linked through the foreign field is not readable by `actor1` the id in this case is. This will be more interesting when doc field access is implemented.

Reply:
> Is there a way to limit inheritance to only the commands that need the identity flag?

Yes, collections do just this - have a look at cli/collections.go

> happy to follow what the consensus is.

I suggest opening up a discord thread, or chatting about it in the standup.

Reply:
> > Is there a way to limit inheritance to only the commands that need the identity flag?
> 
> Yes, collections do just this - have a look at cli/collections.go

Wouldn't it inherit it to `describe` as well? or we can exclude it as well?


> > happy to follow what the consensus is.
> 
> I suggest opening up a discord thread, or chatting about it in the standup.

I am already convinced that only the docIDs that the current identity has access to be should be returned. So unless someone wants to take the opposite position to keep them public it's settled.

- [ ] get docIDs should return only accessible IDs.

Reply:
> Wouldn't it inherit it to describe as well? or we can exclude it as well?

Yes, and PatchCollection, I think the trade-off is worth it given that it is optional, and long-term they'll have ACP anyway

Reply:
- [x] Discussed on discord and made an issue to handle with config file stuff. 

and

- [x] the command to get all docIDs from a collection was implemented to respect identity.

Reply:
All good suggestions tracked here: https://github.com/sourcenetwork/defradb/issues/2416